src/main/java/com/epam/preprod/sirenko/SafeContainer.java

```diff
@@ -0,0 +1,398 @@
+package com.epam.preprod.sirenko;
+
+import com.epam.preprod.sirenko.entity.Product;
+
+import java.util.*;
+
+/**
+ * Resizable array-implementation of the List interface
+ *
+ * @author E.Sirenko
+ **/
+public class SafeContainer<E extends Product> implements List<E> {
```

Comment:
Maybe we should use better naming to indicate behavior, for example CopyOnWriteContainer